compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -688,7 +688,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                 let entry = spanned_predicates.entry(spans);
                                 entry.or_insert_with(|| (path, tr_self_ty, Vec::new())).2.push(p);
                             }
-                            Some(_) => unreachable!(),
+                            Some(_node) => {
```

Comment:
I'm not sure if we should make this a bit more specific matching against auto traits...